.circleci/config.yml

```diff
@@ -307,53 +307,88 @@ jobs:
             kill $TEKU_PID
             exit $EXIT_CODE
       - store_artifacts:
-          path: openapi.json
+          path: ./docs/openapi.json
       - store_artifacts:
           path: teku_output.log
       - persist_to_workspace:
           root: ~/project
           paths:
-            - ./
+            - ./docs/openapi.json
 
   publishAPIDoc:
     executor: node_executor
     steps:
+      # This job requires that we checkout current branch first to have all the repos.
+      # it will not be used directly as we have the saved workspace
+      # but will enable to checkout gh-pages branch.
       - checkout
       - attach_workspace:
-          at: ~/project
-      - add_ssh_keys:
-          fingerprints:
-            - '45:36:58:67:43:15:c6:5f:2c:58:ec:7f:71:e2:e6:ef'
+          at: /tmp/workspace
       - run:
-          name: Set Git user params
+          name: checkout gh pages branch and copy files from workspace
           command: |
-            git config --global user.name $CIRCLE_USERNAME
-            git config --global user.email "${CIRCLE_USERNAME}@users.noreply.github.com"
+            # publishing happens by pushing files to gh-pages branch
+            # retrieve branch to publish to from CIrcle CI env vars and fallback on test-gh-pages
+            # if no var defined
+            [ -n "$BRANCH_OVERRIDE" ] && readonly BRANCH="$BRANCH_OVERRIDE" || readonly BRANCH='test-gh-pages'
+
+            # switch to the publishing branch
+            git checkout "$BRANCH"
+
+            # copy the content of the docs folder built on the previous job and attached
+            # using workspace on temp location
+            # cp with T option to override existing content, specially usefull for latest
+            cp -aT /tmp/workspace/docs/. .
```

Comment:
Should we delete the existing content first?  Otherwise if a file is moved both versions will stay around forever with one being stale.

Reply:
We want to keep the content to preserve all the specs from previous versions. If any changes are made that require to delete some files from gh-pages branch, it would be done by someone knowing the system and who will be able to update the gh-pages branch manually. It should not be a process that will be used often, so I prefer not to build a complex thing and potential bugs for that.